src/main/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/services/OcrDataValidator.java

```diff
@@ -1,23 +1,111 @@
 package uk.gov.hmcts.reform.bulkscanccdeventhandler.services;
 
+import org.springframework.stereotype.Service;
 import uk.gov.hmcts.reform.bulkscanccdeventhandler.model.FormType;
 import uk.gov.hmcts.reform.bulkscanccdeventhandler.model.OcrDataField;
+import uk.gov.hmcts.reform.bulkscanccdeventhandler.model.OcrFieldNames;
 import uk.gov.hmcts.reform.bulkscanccdeventhandler.model.out.OcrValidationResult;
+import uk.gov.hmcts.reform.bulkscanccdeventhandler.model.out.ValidationStatus;
+import uk.gov.hmcts.reform.bulkscanccdeventhandler.util.OcrFormValidationHelper;
 
+import java.util.Arrays;
 import java.util.List;
 
+import static java.util.Arrays.asList;
 import static java.util.Collections.emptyList;
+import static java.util.Collections.singletonList;
+import static uk.gov.hmcts.reform.bulkscanccdeventhandler.model.FormType.CONTACT;
+import static uk.gov.hmcts.reform.bulkscanccdeventhandler.model.FormType.PERSONAL;
+import static uk.gov.hmcts.reform.bulkscanccdeventhandler.model.OcrFieldNames.CONTACT_NUMBER;
+import static uk.gov.hmcts.reform.bulkscanccdeventhandler.model.OcrFieldNames.EMAIL;
+import static uk.gov.hmcts.reform.bulkscanccdeventhandler.model.out.ValidationStatus.ERRORS;
 import static uk.gov.hmcts.reform.bulkscanccdeventhandler.model.out.ValidationStatus.SUCCESS;
+import static uk.gov.hmcts.reform.bulkscanccdeventhandler.model.out.ValidationStatus.WARNINGS;
+import static uk.gov.hmcts.reform.bulkscanccdeventhandler.util.OcrFormValidationHelper.getOcrFieldNames;
+import static uk.gov.hmcts.reform.bulkscanccdeventhandler.util.OcrFormValidationHelper.isValidEmailAddress;
+import static uk.gov.hmcts.reform.bulkscanccdeventhandler.util.OcrFormValidationHelper.isValidPhoneNumber;
 
-public final class OcrDataValidator {
+@Service
+public class OcrDataValidator {
 
-    // prevent instantiation
-    private OcrDataValidator() {
+    public OcrValidationResult validate(FormType formType, List<OcrDataField> ocrData) {
+        List<String> errors = validateMandatoryFields(formType, ocrData);
+        List<String> warnings = validateOptionalFields(formType, ocrData);
+
+        return new OcrValidationResult(warnings, errors, getValidationStatus(!errors.isEmpty(), !warnings.isEmpty()));
+    }
+
+    private List<String> validateMandatoryFields(FormType formType, List<OcrDataField> ocrData) {
+        List<String> mandatoryFields = getMandatoryFieldsForForm(formType);
```

Comment:
At this point if the list is empty you should actually break and no further check would be required as you are not aware of the form type.

Reply:
or alternatively you can check if the input form type is valid upfront and then do your remaining checks ?

Reply:
Surely that's the way to do it. After all, unrecognised form type results with 404.

Reply:
I added a `TODO` to add the `formType` as a path parameter in the endpoint. If formType can't be changed as a path parameter, an exception can be thrown here.